managers/backup-manager/BackupManager.js

```diff
@@ -0,0 +1,421 @@
+          };
+        case CONST.APISERVER.RESOURCE_STATE.ABORTED:
+          return {
+            description: `${action} deployment ${deploymentName} aborted at ${timestamp}`,
+            state: CONST.APISERVER.RESOURCE_STATE.FAILED
+          };
+        case CONST.APISERVER.RESOURCE_STATE.FAILED:
+          return {
+            description: `${action} deployment ${deploymentName} failed at ${timestamp} with Error "${result.stage}"`,
+            state: CONST.APISERVER.RESOURCE_STATE.FAILED
+          };
+        default:
+          return {
+            description: `${action} deployment ${deploymentName} is still in progress: "${result.stage}"`,
+            state: CONST.APISERVER.RESOURCE_STATE.IN_PROGRESS
+          };
+        }
+      });
+  }
+
+  getBackupOperationState(opts) {
+    logger.info('Getting Backup operation State for:', opts);
+    const agent_ip = opts.agent_ip;
+    const options = _.assign({
+      service_id: this.plan.service.id,
+      plan_id: this.plan.id,
+      tenant_id: opts.context ? this.getTenantGuid(opts.context) : opts.tenant_id
+    }, opts);
+
+    function isFinished(state) {
+      return _.includes(['succeeded', 'failed', 'aborted'], state);
+    }
+
+    return this.agent
+      .getBackupLastOperation(agent_ip)
+      .tap(lastOperation => {
+        if (isFinished(lastOperation.state)) {
+          return this.agent
+            .getBackupLogs(agent_ip)
+            .tap(logs => _.each(logs, log => logger.info(`Backup log for: ${opts.instance_guid} - ${JSON.stringify(log)}`)))
+            .then(logs => this.backupStore
+              .patchBackupFile(options, {
+                state: lastOperation.state,
+                logs: logs,
+                snapshotId: lastOperation.snapshotId
+              })
+            ).then(() => this.backupStore.getBackupFile(options))
+            .then(metadata => eventmesh.apiServerClient.updateOperationResponse({
+              resourceId: options.instance_guid,
+              operationName: CONST.APISERVER.ANNOTATION_NAMES.BACKUP,
+              operationType: CONST.APISERVER.ANNOTATION_TYPES.BACKUP,
+              operationId: metadata.backup_guid,
+              value: metadata
+            }));
+        }
+      });
+  }
+
+  deleteBackup(options) {
+    logger.info('Attempting delete with:', options);
+    return this.backupStore
+      .deleteBackupFile(options)
+      .then(() => eventmesh.apiServerClient.updateOperationState({
+        resourceId: options.instance_guid,
+        operationName: CONST.APISERVER.ANNOTATION_NAMES.BACKUP,
+        operationType: CONST.APISERVER.ANNOTATION_TYPES.BACKUP,
+        operationId: options.backup_guid,
+        stateValue: CONST.APISERVER.RESOURCE_STATE.DELETED
+      }))
+      .catch(err => {
+        return Promise
+          .try(() => logger.error(`Error during delete of backup`, err))
+          .tap(() => eventmesh.apiServerClient.updateOperationState({
```

Comment:
Dont think we should update state of delete to error here. Its still a valid backup record which could not be deleted. Just check.